src/utils/hooks/useTextOverflow.ts

```diff
@@ -0,0 +1,47 @@
+import {useState, useEffect, useRef} from "react";
+
+/**
+ * Custom hook to detect whether the text within a referenced container is truncated (ellipsis).
+ * It can be used to determine if a tooltip should be displayed based on text visibility.
+ *
+ * @template T
+ * @param {string} label - The text to monitor for overflow.
+ * @returns {Object} An object containing:
+ * @returns {boolean} isTextTruncated - A boolean indicating if the text is truncated.
+ * @returns {React.RefObject<RefElement>} textRef - A ref to be attached to the text element to measure its dimensions.
```

Comment:
is this the correct way of documenting object return types?

Reply:
Yes you are right, the object return type is not fully correct, it's fixed now ! [5e47cc8](https://github.com/inovex/scrumlr.io/pull/4557/commits/5e47cc8641c9ff0d6cf1b50142445ad03088dfc9)